lib/sparrow/h2_worker/state.ex

```diff
@@ -47,6 +63,7 @@ defmodule Sparrow.H2Worker.State do
   @spec reset_connection_ref(t) :: t
   def reset_connection_ref(state) do
     %__MODULE__{
+      name: :disconnected,
```

Comment:
How exactly do we regain `connected` state? I can't seem to find the code that set's it correctly. 
Also do we really need this state to be that explicit? Is it in any way different then comparing `connection_ref` to `nil`?

Reply:
When the connection is successfully established,  the zSparrow.H2Worker.State.new/3` is called. If `connection_ref=nil`, the state name is set to `:disconnected`, otherwise - `:connected`.

Reply:
Ok, makes sense. Though I still don't see how do we actually use this state. All connection-sensitive jobs look at the `connection_ref` anyway.

Reply:
I introdiced it,  because this PR's goal was to change `H2Worker` into FSM (kind of) with those two states.